src/menu.tsx

```diff
@@ -65,39 +66,38 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
         }
 
         const MARGIN_THRESHOLD = 12;
-        const {top: topTarget, width: widthTarget, left, right, height} = targetRect;
+        const {top: topTarget, width: widthTarget, height} = targetRect;
         const top = topTarget + height;
         const spaceTaken = parseInt(window.getComputedStyle(menu).getPropertyValue('height')) ?? 0;
 
-        const leftDirection = position === 'left' ? left : right - (width ?? widthTarget);
+        const rightDirection = position === 'left' ? 'auto' : `calc(100% - ${widthTarget}px)`;
 
         // if it doesn't fit on bottom
         if (top + spaceTaken + MARGIN_THRESHOLD > window.innerHeight) {
             const availableSpaceBottom = window.innerHeight - top;
             if (topTarget /* this is the available space on top */ > availableSpaceBottom) {
-                const newTop = topTarget - spaceTaken;
                 setItemsComputedProps({
-                    width: widthTarget,
-                    left: leftDirection,
-                    top: Math.max(newTop, MARGIN_THRESHOLD),
-                    maxHeight: topTarget - MARGIN_THRESHOLD,
+                    right: rightDirection,
+                    bottom: '100%',
+                    top: 'auto',
+                    maxHeight: Math.min(topTarget, MAX_HEIGHT_DEFAULT),
                     transformOrigin: 'center bottom',
                 });
             } else {
                 setItemsComputedProps({
-                    width: widthTarget,
-                    top,
-                    left: leftDirection,
+                    top: '100%',
```

Comment:
moved to percentage values, this way we should not think about absolute px values.